src/pages/index.js

```diff
@@ -0,0 +1,291 @@
+const photoModalCaption = photoModal.querySelector(".modal__description");
+
+const profileFormElement = document.forms["edit-profile"];
+const profileSaveBtn = profileFormElement.querySelector(".modal__save-button");
+
+const nameInput = profileFormElement.querySelector("#name");
+const jobInput = profileFormElement.querySelector("#description");
+
+const profileNameElement = document.querySelector(".profile__name");
+const profileJobElement = document.querySelector(".profile__description");
+
+api
+  .getAppInfo()
+  .then((result) => {
+    const savedCards = result[0];
+    const savedProfile = result[1];
+    imageAvatar.src = savedProfile["avatar"];
+    imageAvatar.alt = savedProfile["name"];
+    profileNameElement.textContent = savedProfile["name"];
+    profileJobElement.textContent = savedProfile["about"];
+    savedCards.forEach((card) => {
+      const cardEl = getCardElements(card);
+      cardsList.append(cardEl);
+    });
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", escapeListener);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", escapeListener);
+}
+
+function escapeListener(evt) {
+  const modal = document.querySelector(".modal_opened");
+  if (evt.key === "Escape") {
+    closeModal(modal);
+  }
+}
+
+const modalList = document.querySelectorAll(".modal");
+modalList.forEach((modal) => {
+  modal.addEventListener("click", function (evt) {
+    if (evt.target.parentElement.classList.contains("page")) {
+      closeModal(modal);
+    }
+  });
+});
+
+const closeButtons = document.querySelectorAll(".modal__close-button");
+closeButtons.forEach((button) => {
+  const popup = button.closest(".modal");
+  button.addEventListener("click", (evt) => {
+    if (button.classList.contains("modal__close-button_cancel")) {
+      evt.preventDefault();
+    }
+    closeModal(popup);
+  });
+});
+
+profileEditButton.addEventListener("click", () => {
+  openModal(editModal);
+
+  nameInput.value = profileNameElement.textContent;
+  jobInput.value = profileJobElement.textContent;
+
+  profileSaveBtn.disabled = true;
+  profileSaveBtn.classList.add("modal__save-button_disabled");
+});
+
+// The form submission handler. Note that its name
+// starts with a verb and concisely describes what it does.
+function handleProfileFormSubmit(evt) {
+  // Prevent default browser behavior, see explanation below.
+  evt.preventDefault();
+
+  const newName = nameInput.value;
+  const newJob = jobInput.value;
+  profileSaveBtn.textContent = "Saving...";
+  api
+    .updateProfileInfo({ name: newName, about: newJob })
+    .then((data) => {
+      profileNameElement.textContent = data["name"];
+      profileJobElement.textContent = data["about"];
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      closeModal(editModal);
```

Comment:
` closeModal(editModal);`

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well


**Please, correct it everywhere in the code**